### PR TITLE
Authn unit tests

### PR DESCRIPTION
**Pull request by nela-cc**
### Description
Add unit tests and increase coverage for services/authn/user_handler.go and impl/authn/authn.go.
Rename other `handlers` file to match convention -> `handler`.

### Relevant Issues/PRs
Issue #239

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [x] Added tests
- [ ] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [x] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [ ] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [x] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: encryption-service/impl/authn/authn_test.go [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+func TestNewGroup(t *testing.T) {
+	userAuthenticator, err := SetupUA()
+	if err != nil {
+		t.Fatalf("NewGroup errored: %s", err)
+	}
+
+	authStoreTx := &authstorage.AuthStoreTxMock{
+		InsertGroupFunc: func(ctx context.Context, protected *common.ProtectedGroupData) error {
+			return nil
```
Comment (philip-cc) [inline encryption-service/impl/authn/authn_test.go:288]: We should check if this is actually called.

File: encryption-service/impl/authn/authn_test.go [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+				}
+				groupDataBatch = append(groupDataBatch, *protected)
+			}
+			return groupDataBatch, nil
+		},
+	}
+	ctx := context.WithValue(context.Background(), common.AuthStorageTxCtxKey, authStoreTx)
+
+	groupID := uuid.Must(uuid.NewV4())
+	groupIDs := []uuid.UUID{groupID}
```
Comment (philip-cc) [inline encryption-service/impl/authn/authn_test.go:331]: We should probably test this with more than one group ID.

File: encryption-service/services/authn/user_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+func TestCreateUser(t *testing.T) {
+	outputUserID := uuid.Must(uuid.NewV4())
+	password := "Password"
+
+	userAuthenticator := &authnimpl.UserAuthenticatorMock{
+		NewUserFunc: func(ctx context.Context) (*uuid.UUID, string, error) {
+			return &outputUserID, password, nil
+		},
+		UpdateUserFunc: func(ctx context.Context, userID uuid.UUID, userData *common.UserData) error {
+			return nil
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/services/authn/user_handler_test.go:50]: Done, userID and userData checked.

File: encryption-service/services/authn/user_handler_test.go [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+		},
+		GetUserDataFunc: func(ctx context.Context, userID uuid.UUID) (*common.UserData, error) {
+			return userData, nil
+		},
+		NewGroupWithIDFunc: func(ctx context.Context, groupID uuid.UUID, scopes common.ScopeType) error {
+			return nil
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/services/authn/user_handler_test.go:56]: Done, groupID and scopes checked.

File: encryption-service/services/authn/user_handler_test.go [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+	}
+	if errStatus, _ := status.FromError(err); codes.Internal != errStatus.Code() {
+		t.Fatalf("Wrong error returned: expected %v, but got %v", codes.Internal, errStatus)
+	}
+}
+
+func TestRemoveUser(t *testing.T) {
+	userAuthenticator := &authnimpl.UserAuthenticatorMock{
+		RemoveUserFunc: func(ctx context.Context, userID uuid.UUID) error {
+			return nil
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/services/authn/user_handler_test.go:227]: Done, userID checked.

File: encryption-service/impl/authn/authn_test.go (hunk 2 above)
Comment (nela-cc) [reply encryption-service/impl/authn/authn_test.go:331]: Added a second group ID.

File: encryption-service/impl/authn/authn_test.go (hunk 1 above)
Comment (nela-cc) [reply encryption-service/impl/authn/authn_test.go:288]: Done.

File: encryption-service/impl/authn/authn_test.go [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+
+func TestNewGroupWithID(t *testing.T) {
+	userAuthenticator, err := SetupUA()
+	if err != nil {
+		t.Fatalf("NewGroupWithID errored: %s", err)
+	}
+
+	authStoreTx := &authstorage.AuthStoreTxMock{
+		InsertGroupFunc: func(ctx context.Context, protected *common.ProtectedGroupData) error {
+			return nil
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/impl/authn/authn_test.go:269]: Done.

File: encryption-service/impl/authn/authn_test.go [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+
+func TestRemoveUserFromGroup(t *testing.T) {
+	userAuthenticator, err := SetupUA()
+	if err != nil {
+		t.Fatalf("RemoveUser errored: %s", err)
+	}
+
+	authStoreTx := &authstorage.AuthStoreTxMock{
+		RemoveUserFunc: func(ctx context.Context, userID uuid.UUID) error {
+			return nil
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/impl/authn/authn_test.go:252]: Done.

File: encryption-service/impl/authn/authn_test.go [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+
+func TestUpdateUser(t *testing.T) {
+	userAuthenticator, err := SetupUA()
+	if err != nil {
+		t.Fatalf("UpdateUser errored: %s", err)
+	}
+
+	authStoreTx := &authstorage.AuthStoreTxMock{
+		UpdateUserFunc: func(ctx context.Context, protected *common.ProtectedUserData) error {
+			return nil
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/impl/authn/authn_test.go:115]: Done.

File: encryption-service/impl/authn/authn_test.go [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+					WrappedKey: wrappedKey,
+				}
+				groupDataBatch = append(groupDataBatch, *protected)
+			}
+			return groupDataBatch, nil
+		},
+	}
+	ctx := context.WithValue(context.Background(), common.AuthStorageTxCtxKey, authStoreTx)
+
+	_, err = userAuthenticator.LoginUser(ctx, userID, password)
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/impl/authn/authn_test.go:212]: Done.

File: encryption-service/services/authn/user_handler_test.go [hunk 10]
```diff
@@ -0,0 +1,335 @@
...
+		t.Fatalf("Wrong error returned: expected %v, but got %v", codes.Internal, errStatus)
+	}
+}
+
+func TestLoginUser(t *testing.T) {
+	outputToken := "token"
+
+	userAuthenticator := &authnimpl.UserAuthenticatorMock{
+		LoginUserFunc: func(ctx context.Context, userID uuid.UUID, password string) (string, error) {
+			return outputToken, nil
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/services/authn/user_handler_test.go:219]: Done, userID and password checked.

File: encryption-service/impl/authn/authn_test.go [hunk 11]
```diff
@@ -0,0 +1,339 @@
...
+
+func TestNewUser(t *testing.T) {
+	userAuthenticator, err := SetupUA()
+	if err != nil {
+		t.Fatalf("NewUser errored: %s", err)
+	}
+
+	authStoreTx := &authstorage.AuthStoreTxMock{
+		InsertUserFunc: func(ctx context.Context, protected *common.ProtectedUserData) error {
+			return nil
```
[the remark this answers is not in the document]
Comment (nela-cc) [reply encryption-service/impl/authn/authn_test.go:100]: Done.

**Review by philip-cc — APPROVED** [review]

**Pull request merged**